app/test_main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+import pytest
+from unittest.mock import patch
+from app.main import can_access_google_page
+
+
+@pytest.fixture
+def mock_valid_google_url() -> "patch":
+    with patch("app.main.valid_google_url") as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_has_internet_connection() -> "patch":
+    with patch("app.main.has_internet_connection") as mock:
+        yield mock
```

Comment:
There is no need to create two separate functions, use  `@mock.patch` decorator and apply it to the `test_can_access_google_page` twice, e.g. `@mock.patch("app.main.valid_google_url")`. You could process it as is in the signature of the `test_can_access_google_page`.